react/components/FacetItem.js

```diff
@@ -26,38 +27,58 @@ const FacetItem = ({
   className,
   preventRouteChange,
 }) => {
-  const { showFacetQuantity } = useContext(SettingsContext)
-  const [selected, setSelected] = useState(facet.selected)
-  const handles = useCssHandles(CSS_HANDLES)
+  const { showFacetQuantity } = useContext(SettingsContext);
+  const [selected, setSelected] = useState(facet.selected);
+  const { push } = usePixel();
+  const handles = useCssHandles(CSS_HANDLES);
   const classes = classNames(
     applyModifiers(handles.filterItem, facet.value),
     { [`${handles.filterItem}--selected`]: facet.selected },
     className,
-    'lh-copy w-100'
-  )
+    "lh-copy w-100"
+  );
+  const { searchQuery } = useSearchPage();
 
   const checkBoxId = reservedVariableNames.includes(facet.value)
     ? `filterItem--${facet.key}-${facet.value}`
-    : `${facet.key}-${facet.value}`
+    : `${facet.key}-${facet.value}`;
 
   // This effect fixes the issue described in this PR
   // https://github.com/vtex-apps/search-result/pull/422
   useEffect(() => {
     if (facet.selected !== selected) {
-      setSelected(facet.selected)
+      setSelected(facet.selected);
     }
     // however, having `selected` as a dependency causes it
     // to always reset back to `facet.selected`. So, we remove it,
     // so only changes in facet.selected affect the state.
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [facet.selected])
+  }, [facet.selected]);
+
+  const getCategoryFromObjs = (products) => {
+    const categoryId = products[0].categoryId;
+    const result = products.filter((obj) => obj.categoryId !== categoryId);
+    return result.length == 0 ? categoryId : "";
+  };
 
-  const shouldDisable = useShouldDisableFacet(facet)
+  const pushPixelEvent = (name, value, isSelected) => {
+    if (isSelected) {
+      push({
+        event: "filterManipulation",
+        items: {
+          filterProductCategory: getCategoryFromObjs(searchQuery.products),
+          filterName: name,
+          filterValue: value,
+        },
+      });
+    }
+  };
```

Comment:
Export these functions in one file so you can import them and reuse them on the other files.

Reply:
Done in last commit (required operations)